### PR TITLE
New Zypper Cheat Sheet Goodie

### PR DESCRIPTION
**Pull request by akaysh**
Instant Answer URL
https://duck.co/ia/view/zypper_package_manager_cheat_sheet

### CODE REVIEW

**Comment by daxtheduck** [comment]
@akaysh Hey!
        
Thanks for taking the time to contribute! We really appreciate it.

We work closely with every contributor to make Instant Answers the best they can be, so we appreciate your patience as we look over your code. From here, the process usually goes like this:

1. Pull Request is reviewed by the DuckDuckGo staff and community
2. Staff and community will leave feedback with any necessary updates to the function or design.
3. Once you've made any necessary corrections, then your Instant Answer will be merged and deployed live on DuckDuckGo!

While the team and the community review your submission, we'd like to make sure you look good! If you'd like a free DuckDuckGo tshirt, please send over the following info to merch@duckduckgo.com, and be sure to put "Dev Merch" in the subject line:

Name:
Street Address:
City:
State / Province:
Country:
Phone (Non-US Orders only): 
T-shirt size:
GitHub username:
Link to this Pull Request:

If you have any questions along the way, feel free to ask them here. Our staff and community are also available on Slack to answer any questions you may have. If you'd like to join us there please send an email to <quackslack@duckduckgo.com>.

>More Info: https://duck.co/duckduckhack/submission_and_review

Thanks!

**Comment by daxtheduck** [comment]
Metadata placeholder post
            
*If you haven't already, please edit your first comment to include a link to the IA page at the bottom.*
            
---
            
**How to find your IA page**
- New Instant Answer
  -- follow the steps [here](https://duck.co/ia/new_ia) to create your IA page.
              
- Existing Instant Answer
  -- find your IA page from this [list](https://duck.co/ia)

File: share/goodie/cheat_sheets/json/zypper.json [hunk 1]
```diff
@@ -0,0 +1,94 @@
+{
+    "id": "zypper_cheat_sheet",
+    "name": "Zypper",
+    "description": "Zypper is a command line tool for package management in OpenSUSE and SUSE Enterprise Linux platforms.",
```
Comment (mohan08p) [inline share/goodie/cheat_sheets/json/zypper.json]: Replace the word Zypper with It

File: share/goodie/cheat_sheets/json/zypper.json [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+        "Quick Reference",
+        "Repository Management",
+        "Package Management",
+        "Scripts and Applications"
+    ],
+
+    "sections": {
+        "Quick Reference": [{
+            "key": "zypper",
+            "val": "to print the list of available global options and commands"
```
Comment (mohan08p) [inline share/goodie/cheat_sheets/json/zypper.json]: Start val fields with capitals

File: share/goodie/cheat_sheets/json/zypper.json [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+        }, {
+            "key": "zypper help \\[command\\]",
+            "val": "to print help for the given command"
```
Comment (mohan08p) [inline share/goodie/cheat_sheets/json/zypper.json]: Also, subsequent fields with caps..

File: share/goodie/cheat_sheets/json/zypper.json [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+    "name": "Zypper",
+    "description": "Zypper is a command line tool for package management in OpenSUSE and SUSE Enterprise Linux platforms.",
+
+    "metadata": {
+        "sourceName": "OpenSUSE",
+        "sourceUrl": "https://en.opensuse.org/SDB:Zypper_usage"
+    },
+
+    "aliases": [
+        "OpenSUSE package manager", "apt-get alternative OpenSUSE"
```
Comment (navneetzz) [inline share/goodie/cheat_sheets/json/zypper.json]: @akaysh don't use capitals in aliases. So replace OpenSUSE with opensuse.

File: share/goodie/cheat_sheets/json/zypper.json [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+                "key": "zypper install \\[package-name\\] or zypper in \\[package-name\\]",
+                "val": "installing packages"
+
+            }, {
+                "key": "zypper remove \\[package-name\\] or zypper rm \\[package-name\\]",
+                "val": "removing packages"
+
+            }, {
+                "key": "zypper source-install \\[package-name\\] or zypper si \\[package-name\\]",
+                "val": "source packages and build dependencies"
```
Comment (navneetzz) [inline share/goodie/cheat_sheets/json/zypper.json]: @akaysh capitalize the first letter in value field.

File: share/goodie/cheat_sheets/json/zypper.json [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+                "val": "renaming repositories"
+            }, {
+                "key": "zypper repos --export \\[repo-path\\] or zypper lr -e \\[repo-path\\]",
+                "val": "Import/Export Repositories"
+            }
+
+        ],
+        "Package Management": [{
+                "key": "zypper install \\[package-name\\] or zypper in \\[package-name\\]",
+                "val": "installing packages"
```
Comment (mohan08p) [inline share/goodie/cheat_sheets/json/zypper.json]: I guess, even no need of ing i.e. continuous state... just "Install package" is better. Also, for other val fields...

File: share/goodie/cheat_sheets/json/zypper.json (hunk 3 above)
Comment (akaysh) [inline share/goodie/cheat_sheets/json/zypper.json]: anything apart from val fields?

**Comment by daxtheduck** [comment]
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=zypper cheat sheet)
                
                [Mobile screenshot](https://ia-screenshots.s3.amazonaws.com/zypper_package_manager_cheat_sheet_mobile.png)

File: share/goodie/cheat_sheets/json/zypper.json [hunk 7]
```diff
@@ -0,0 +1,93 @@
...
+            "key": "zypper lp",
+            "value": "See what patch updates are needed"
+        }, {
+            "key": "zypper patch",
+            "value": "Apply the needed patches"
+        }, {
+            "key": "zypper se \\[package-name\\]",
+            "value": "Search for the given package"
+        }, {
+            "key": "zypper rm [package-name\\]",
```
Comment (jophab) [inline share/goodie/cheat_sheets/json/zypper.json]: @akaysh add \\ before [

File: share/goodie/cheat_sheets/json/zypper.json [hunk 8]
```diff
@@ -0,0 +1,93 @@
+{
+    "id": "zypper_cheat_sheet",
+    "name": "Zypper",
+    "description": "It is a command line tool for package management in OpenSUSE and SUSE Enterprise Linux platforms.",
```
Comment (jophab) [inline share/goodie/cheat_sheets/json/zypper.json]: @akaysh Lets remove the period at the end of description

**Comment by daxtheduck** [comment]
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=zypper cheat sheet)
                
                [Mobile screenshot](https://ia-screenshots.s3.amazonaws.com/zypper_package_manager_cheat_sheet_mobile.png)

**Comment by daxtheduck** [comment]
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=zypper cheat sheet)
                
                [Mobile screenshot](https://ia-screenshots.s3.amazonaws.com/zypper_package_manager_cheat_sheet_mobile.png)

**Comment by daxtheduck** [comment]
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=zypper cheat sheet)
                
                [Mobile screenshot](https://ia-screenshots.s3.amazonaws.com/zypper_package_manager_cheat_sheet_mobile.png)

File: share/goodie/cheat_sheets/json/zypper.json [hunk 9]
```diff
@@ -0,0 +1,93 @@
...
+            "value": "Remove the given package"
+        }, {
+            "key": "zypper in \\[package-name\\]",
+            "value": "Install the given package"
+        }, {
+            "key": "zypper up \\[package-name\\]",
+            "value": "Update the given or all installed packages with newer versions, where possible"
+        }],
+        "Repository Management": [{
+                "key": "zypper repos \\[repo-name\\] or zypper lr \\[repo-name\\]",
```
Comment (moollaza) [inline share/goodie/cheat_sheets/json/zypper.json]: It look like all of these have an `or`? It's making the content a little lengthier than it needs to be.

Let's indicate the alternate in with some syntax: `zypper (refresh|rf) \\[repo-name\\]` -- do the man pages offer a more succinct way of display the alternatives?

We could also wrap each command in a `[]` which will force them to be displayed in separate code blocks.

**Comment by daxtheduck** [comment]
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=zypper cheat sheet)
                
                [Mobile screenshot](https://ia-screenshots.s3.amazonaws.com/zypper_package_manager_cheat_sheet_mobile.png)

File: share/goodie/cheat_sheets/json/zypper.json (hunk 9 above)
Comment (moollaza) [inline share/goodie/cheat_sheets/json/zypper.json]: @akaysh Ping! I'm just checking in to see if you're still interested in finishing this one up? We'd love to see this go live!

There's absolutely no rush, just wanted to make sure you're not stuck on anything.

**Comment by daxtheduck** [comment]
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=zypper cheat sheet)
                
                [Mobile screenshot](https://ia-screenshots.s3.amazonaws.com/zypper_package_manager_cheat_sheet_mobile.png)

**Comment by moollaza** [comment]
@akaysh I'm merging this. It would great if you could add in the suggested changes to make this a little easier to read :+1:

**Comment by daxtheduck** [comment]
@akaysh this is now merged! It should be live on DuckDuckGo within one week (or during the next release) and we'll be sure to notify you when it happens.

After your Instant Answer goes live for the World to use, you will be able to see [traffic information](http://docs.duckduckhack.com/maintaining/tools.html#monitor-your-instant-answer-page) about your IA as well as receive feedback directly from users. 
                
Please refer to the [maintenance guide and docs](http://docs.duckduckhack.com/maintaining/guidelines.html) to keep up with potential changes to your Instant Answer.
                
Thank you again for your help!

**Pull request merged**